tests/storage/business/test_watcher.py

```diff
@@ -112,6 +173,100 @@ def test_scan(tmp_path: Path):
     assert call.args[1] is None
 
 
+@pytest.mark.unit_test
+def test_scan_recursive_false(tmp_path: Path, db_session: Session):
+    # Base.metadata.create_all(db_session)
+
+    def count_studies():
+        return db_session.execute("SELECT COUNT(*) FROM study").scalar()
+
+    clean_files()
+
+    """
+    Create this hierarchy
+
+    tmp_path
+    ├── default
+    │   └── studyA
+    │       └── study.antares
+    └── diese
+        └── folder
+            ├── studyC
+            │   └── study.antares
+            ├── trash
+            │   └── trash
+            ├── another_folder
+            │   ├── AW_NO_SCAN
+            │   └── study.antares
+            └── subfolder
+                └── studyG
+                    └── study.antares   
+    """
+    default = tmp_path / "default"
```

Comment:
Perhaps I'm missing something but you could use the fixture you've introduced no ?

Reply:
The fixture was in the test for the explorer. 

I could create a fixture in test_watcher too, the folder structure isn't exactly identical in  test_scan and test_scan_recursive_false but you think it's worth it to centralize it ?

Reply:
Okay. I think it's worth centralizing if you can.